src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -12450,87 +12401,82 @@ namespace SurfaceGeometry {
+                if (thisSurface.Class != SurfaceClass::Wall && thisSurface.Class != SurfaceClass::Floor && thisSurface.Class != SurfaceClass::Roof) {
                     ++notused;
                     surfacenotused(notused) = SurfNum;
                     continue;
                 }
 
                 ++NActFaces;
-                ZoneStruct.SurfaceFace(NActFaces).FacePoints.allocate(state.dataSurface->Surface(SurfNum).Sides);
-                ZoneStruct.SurfaceFace(NActFaces).NSides = state.dataSurface->Surface(SurfNum).Sides;
-                ZoneStruct.SurfaceFace(NActFaces).SurfNum = SurfNum;
-                ZoneStruct.SurfaceFace(NActFaces).FacePoints({1, state.dataSurface->Surface(SurfNum).Sides}) =
-                    state.dataSurface->Surface(SurfNum).Vertex({1, state.dataSurface->Surface(SurfNum).Sides});
-                CreateNewellAreaVector(ZoneStruct.SurfaceFace(NActFaces).FacePoints,
-                                       ZoneStruct.SurfaceFace(NActFaces).NSides,
-                                       ZoneStruct.SurfaceFace(NActFaces).NewellAreaVector);
-                SumAreas += VecLength(ZoneStruct.SurfaceFace(NActFaces).NewellAreaVector);
+                auto &thisFace = ZoneStruct.SurfaceFace(NActFaces);
+                thisFace.FacePoints.allocate(thisSurface.Sides);
+                thisFace.NSides = thisSurface.Sides;
+                thisFace.SurfNum = SurfNum;
+                thisFace.FacePoints({1, thisSurface.Sides}) = thisSurface.Vertex({1, thisSurface.Sides});
+                Vectors::CreateNewellAreaVector(thisFace.FacePoints, thisFace.NSides, thisFace.NewellAreaVector);
+                SumAreas += Vectors::VecLength(thisFace.NewellAreaVector);
             }
             ZoneStruct.NumSurfaceFaces = NActFaces;
-            SurfCount = double(NActFaces);
 
-            bool isFloorHorizontal;
-            bool isCeilingHorizontal;
-            bool areWallsVertical;
+            bool isFloorHorizontal = false;
+            bool isCeilingHorizontal = false;
+            bool areWallsVertical = false;
             std::tie(isFloorHorizontal, isCeilingHorizontal, areWallsVertical) = areSurfaceHorizAndVert(state, ZoneStruct);
-            Real64 oppositeWallArea;
-            Real64 distanceBetweenOppositeWalls;
+            Real64 oppositeWallArea = 0.0;
+            Real64 distanceBetweenOppositeWalls = 0.0;
 
             bool areWallsSameHeight = areWallHeightSame(state, ZoneStruct);
 
             std::vector<EdgeOfSurf> listOfedgeNotUsedTwice;
             bool isZoneEnclosed = isEnclosedVolume(ZoneStruct, listOfedgeNotUsedTwice);
             ZoneVolumeCalcMethod volCalcMethod;
 
+            Real64 floorAreaForVolume = (thisZone.FloorArea > 0.0) ? thisZone.FloorArea : thisZone.geometricFloorArea;
+            Real64 ceilingAreaForVolume = (thisZone.CeilingArea > 0.0) ? thisZone.CeilingArea : thisZone.geometricCeilingArea;
```

Comment:
For volume calcs, use geometric area if area based on HT surfs or user input is zero.